test/libyul/yulSyntaxTests/hex_string_literal_non_printable_characters.yul

```diff
@@ -0,0 +1,6 @@
+{
+    let name := hex"000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"
+}
```

Comment:
This was not failing (only assembly blocks were affected) but we did not have this case covered so I'm adding a test.